### ISSUE TITLE
Hash type problems

### ISSUE THREAD

**Issue by mrrcollins** [glarizza/puppet-property_list_key#7]
I have this in a manifest:

```
       property_list_key { 'afp-loggingAttributes':
                ensure     => present,
                path       => '/Library/Preferences/com.apple.AppleFileServer.plist',
                key        => 'loggingAttributes',
                value      => {'logCreateDir' => true, 'logCreateFile' => true, 'logDelete' => true, 'logLogin' => true, 'logLogout' => true, 'logOpenFork' => false},
                value_type => 'hash',
        }
```

If loggingAttributes isn't there, it creates everything correctly:

```
    <key>loggingAttributes</key>
    <dict>
        <key>logCreateDir</key>
        <true/>
        <key>logCreateFile</key>
        <true/>
        <key>logDelete</key>
        <true/>
        <key>logLogin</key>
        <true/>
        <key>logLogout</key>
        <true/>
        <key>logOpenFork</key>
        <false/>
    </dict>
```

But if I go to change it, the entire dictionary becomes an array:

```
       property_list_key { 'afp-loggingAttributes':
                ensure     => present,
                path       => '/Library/Preferences/com.apple.AppleFileServer.plist',
                key        => 'loggingAttributes',
                value      => {'logCreateDir' => true, 'logCreateFile' => true, 'logDelete' => true, 'logLogin' => true, 'logLogout' => true, 'logOpenFork' => true},
                value_type => 'hash',
        }
```

After:

```
    <key>loggingAttributes</key>
    <array>
        <dict>
            <key>logCreateDir</key>
            <true/>
            <key>logCreateFile</key>
            <true/>
            <key>logDelete</key>
            <true/>
            <key>logLogin</key>
            <true/>
            <key>logLogout</key>
            <true/>
            <key>logOpenFork</key>
            <true/>
        </dict>
    </array>
```

And puppet changes it everytime it runs after this.

**Comment by nybblr** [comment]
@mrrcollins had this same issue. I've tracked down the issue to the following line:
https://github.com/glarizza/puppet-property_list_key/blob/master/lib/puppet/provider/property_list_key/rubycocoa.rb#L89

The `:hash` should not be included in that particular when statement; if it is removed completely, everything works because the default grabs `.first` which "unwraps" the default array type into the hash expected.

I'm forking this now with the fix. Will post shortly.

~ Jonathan

**Comment by glarizza** [comment]
Fixed with #10 - thanks for reporting!

**Comment by nybblr** [comment]
Sweet, thanks @glarizza
